stdlib/public/core/Arrays.swift.gyb

```diff
@@ -2087,6 +2087,72 @@ public func != <Element : Equatable>(
 ) -> Bool {
   return !(lhs == rhs)
 }
+
+extension ${Self} {
+  /// Calls a closure with a view of the array's underlying bytes of memory as a
+  /// Collection of `UInt8`.
+  /// ${contiguousCaveat}
+  ///
+  /// - Precondition: `Pointee` is a trivial type.
+  ///
+  /// The following example shows how you copy bytes into an array:
+  ///
+  ///    var numbers = [Int32](repeating: 0, count: 2)
+  ///    var byteValues: [UInt8] = [0x01, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00]
+  ///    numbers.withUnsafeMutableBytes { destBytes in
+  ///      byteValues.withUnsafeBytes { srcBytes in
+  ///        destBytes.copyBytes(from: srcBytes)
+  ///      }
+  ///    }
+  ///
+  /// - Parameter body: A closure with an `UnsafeRawBufferPointer` parameter
+  ///   that points to the contiguous storage for the array. If `body` has a
+  ///   return value, it is used as the return value for the
+  ///   `withUnsafeBytes(_:)` method. The argument is valid only for the
+  ///   duration of the closure's execution.
+  /// - Returns: The return value of the `body` closure parameter, if any.
+  ///
+  /// - SeeAlso: `withUnsafeBytes`, `UnsafeMutableRawBufferPointer`
+  public mutating func withUnsafeMutableBytes<R>(
+    _ body: (inout UnsafeMutableRawBufferPointer) throws -> R
+  ) rethrows -> R {
+    return try self.withUnsafeMutableBufferPointer {
+      var bytes = UnsafeMutableRawBufferPointer($0)
```

Comment:
Oh... this is actually really interesting.

The closure argument was 'inout' in order to allow the closure body to invoke a subscript-by-range setter. This was a bit of nonsense inherited from UnsafeMutableBufferPointer, which requires the subscript setter to be mutating. I think it should be nonmutating (because a pointer has reference semantics):

```
  public subscript(bounds: Range<Int>) -> UnsafeMutableRawBufferPointer {
    nonmutating set { ... }
  }
```

This way withUnsafeBytes() does not need to take an `inout` argument. This is an easy fix for raw buffer pointer. We may want to later revisit whether typed buffer pointer's subscript-by-range should also be nonmutating.